lib/icingadb/icingadb.cpp

```diff
@@ -81,19 +79,36 @@ void IcingaDB::Start(bool runtimeCreated)
 
 	m_WorkQueue.SetExceptionCallback([this](boost::exception_ptr exp) { ExceptionHandler(std::move(exp)); });
 
-	m_Rcon = new RedisConnection(GetHost(), GetPort(), GetPath(), GetUsername(), GetPassword(), GetDbIndex(),
-		GetEnableTls(), GetInsecureNoverify(), GetCertPath(), GetKeyPath(), GetCaPath(), GetCrlPath(),
-		GetTlsProtocolmin(), GetCipherList(), GetConnectTimeout(), GetDebugInfo());
+	RedisConnInfo::ConstPtr connInfo = new RedisConnInfo(
+		GetEnableTls(),
+		GetInsecureNoverify(),
+		GetPort(),
+		GetDbIndex(),
+		GetConnectTimeout(),
+		GetHost(),
+		GetPath(),
+		GetUsername(),
+		GetPassword(),
+		GetCertPath(),
+		GetKeyPath(),
+		GetCaPath(),
+		GetCrlPath(),
+		GetTlsProtocolmin(),
+		GetCipherList(),
+		GetDebugInfo()
+	);
```

Comment:
I'd rather initialize the field individually, i.e.:

```c++
RedisConnInfo::Ptr connInfo = new RedisConnInfo();
connInfo.EnableTls = GetEnableTls();
// ...
```

Then you can check whether the values are passed correctly here without needing IDE annotations (or very carefully checking them one by one) and also can just omit the following constructor:
https://github.com/Icinga/icinga2/blob/ac841b360c229a4d14f805ad289ac89420d72cec/lib/icingadb/redisconnection.hpp#L76-L110

Reply:
Why was the [nice initialization syntax (`{.x = 42}`)](https://en.cppreference.com/w/cpp/language/aggregate_initialization.html) only added in C++20 (when it's already in C99)? :(

Anyway, I think we could still get a reasonable level of const-correctness if there was an `RedisConnInfo::ConstPtr IcingaDB::GetRedisConnInfo()` that does the initialization and is the only place where there's any non-const use of `RedisConnInfo`.